lib/models/device.coffee

```diff
@@ -27,6 +28,37 @@ configModel = require('./config')
 applicationModel = require('./application')
 auth = require('../auth')
 
+MIN_SUPERVISOR_APPS_API = '1.7.1'
+
+###*
+# @summary Check supervisor version compatibility using semver
+# @name isCompatible
+# @private
+# @function
+# @memberof resin.models.device
+#
+# @param {String} version - version under check
+# @param {String} minVersion - minimum accepted version
+# @fulfil {Boolean} - is compatible
+# @reject {Error} Will reject if the given version is < than the given minimum version
+# @returns {Promise}
+#
+# @example
+# resin.models.device.isCompatible(version, MIN_VERSION).then(function(isCompatible) {
+# 	console.log(isCompatible);
+# });
+#
+# @example
+# resin.models.device.isCompatible(version, MIN_VERSION, function(error, isCompatible) {
+# 	if (error) throw error;
+# 	console.log(isCompatible);
+# });
+###
+exports.isCompatible = Promise.method (version, minVersion) ->
```

Comment:
The `export` was made with the assumption that this method could potentially be reused by other modules - like, for instance, `getPubNubKeys` from the `config` model, which is also private.

I agree that `is` implies a `true` or `false` return value, yet I tried to use a short function name and couldn't really think of something better - the function's documentation also clearly states that the method refers to the supervisor version. 

I'm pushing a change with the proposed `ensureSupervisorCompability` name to be consistent with the SDK coding conventions.